src/services/download/mod.rs

```diff
@@ -128,14 +130,30 @@ impl DownloadService {
         hub: Arc<Hub>,
     ) -> impl Future<Output = Result<Vec<SourceFileId>, DownloadError>> {
         let worker = self.worker.clone();
+
+        // The Sentry cache index should expire as soon as we attempt to retry negative caches.
+        let sentry_index_cache_ttl = if self.config.cache_dir.is_some() {
+            self.config.caches.downloaded.retry_misses_after
+        } else {
+            None
+        };
```

Comment:
This is an implementation detail of the sentry source. Can you move this into the `sentry` source and rather pass the entire config in?

@flub this is probably another good reason to make sources `structs`. That way, we get rid of `lazy_static`s and we can hold on to a config reference so we don't have to pipe arbitrary attributes through multiple methods (e.g. having the TTL on `SearchQuery` is odd).

Reply:
ok done. this clones the arc unfortunately. we use async and std futures, but unfortunately we need to guarantee that those are `'static` because we convert them into old futures somewhere, I think. I'll attempt this separately

Reply:
@jan-auer not sure i follow, you mean `SourceConfig` should be a struct and you want to attach the caches to that?